network/rpc/server/src/bin/client.rs

```diff
@@ -0,0 +1,82 @@
+//! Example of using jsonrpsee client.
+//!
+//! Create a nexus project, build it and get path to elf file, for example
+//! ```sh
+//! realpath target/riscv32i-unknown-none-elf/debug/my_program
+//! ```
+//!
+//! Run the client with `cargo run -r --bin client -- <PATH TO ELF>`
+
+use std::env;
+use std::time::Duration;
+
+use jsonrpsee::ws_client::{WsClient, WsClientBuilder};
+use nexus_prover::types::IVCProof;
+use nexus_rpc_common::ArkWrapper;
+use nexus_rpc_traits::RpcClient;
+
+use tracing::Level;
+use tracing_subscriber::{
+    filter, fmt::format::FmtSpan, layer::SubscriberExt, util::SubscriberInitExt,
+};
+
+// type used by the RPC.
+type RpcProof = ArkWrapper<IVCProof>;
+
+#[tokio::main]
+async fn main() {
+    setup_logger();
+
+    // path to elf
+    let args: Vec<String> = env::args().collect();
+    if args.len() != 2 {
+        panic!("invalid number of arguments")
+    }
+    let path = &args[1];
+
+    let elf_bytes = std::fs::read(path).expect("failed to read elf file");
+    let proof = request(elf_bytes).await;
+
+    let pp = nexus_rpc_server::load_params();
+    proof
+        .verify(&pp, proof.step_num() as usize)
+        .expect("proof is invalid");
+
+    tracing::info!(
+        target: "nexus-rpc-test-client",
+        "Proof is valid!"
+    );
+}
+
+async fn request(elf_bytes: Vec<u8>) -> RpcProof {
```

Comment:
nit: Using the `ElfBytes` type alias from the common crate would be more consistent.